qds_sdk/commands.py

```diff
@@ -254,6 +254,98 @@ def parse(cls, args):
+	
+    optparser.add_option("--notify", action="store_true", dest="can_notify", default=False, help="sends an email on command completion")
+
+    optparser.add_option("--name", dest="name", help="Assign a name to this query")
+           
+
+    @classmethod
+    def parse(cls, args):
+        """
+        Parse command line arguments to construct a dictionary of command
+        parameters that can be used to create a command
+
+        Args:
+            `args`: sequence of arguments
+
+        Returns:
+            Dictionary that can be used in create method
+
+        Raises:
+            ParseError: when the arguments are not correct
+        """
+        allowedlanglist = ["python", "scala"]
+        try:
+            (options, args) = cls.optparser.parse_args(args)
+            if options.program is None and options.script_location is None and options.commandline is None:
+                raise ParseError("One of script location or program or commandline must be specified", cls.optparser.format_help())
+        except OptionParsingError as e:
+            raise ParseError(e.msg, cls.optparser.format_help())
+        except OptionParsingExit as e:
+            return None
+
+        if options.script_location is not None:
+            if options.program is not None:
+                if options.commandline is not None:
+                    raise ParseError("All the three parameters program, script_location, and cmdline cannot be specified together", cls.optparser.format_help())
+                else:
+                    raise ParseError("Both Program and script_location cannot be specified", cls.optparser.format_help())
+            elif options.commandline is not None:
+                    raise ParseError("Both commandline and script_location cannot be specified together", cls.optparser.format_help())
+            if ((options.script_location.find("s3://") != 0) and
+                (options.script_location.find("s3n://") != 0)):
+
+                # script location is local file
+
+                try:
+                    q = open(options.script_location).read()
+                except IOError as e:
+                    raise ParseError("Unable to open script location: %s" %
+                                     str(e),
+                                     cls.optparser.format_help())
+                options.script_location = None
+                options.query = q
+        elif options.program is not None:
+            if options.commandline is not None:
+                raise ParseError("Both commandline and program cannot be specified together", cls.optparser.format_help())
+        if options.commandline is not None or options.script_location is not None:
+            if options.language is not None:
+                raise ParseError("Both commandline/script location and language cannot be specified together", cls.optparser.form)
+        elif options.language not in allowedlanglist:
```

Comment:
To frame it in a better way. When is the language options required?
For which of the the three input options is it necessary or optional?